### PR TITLE
Attachments

### PR DESCRIPTION
**Pull request by asmega**
- Part 1 of a few changes
- When the async job runs we create an identifier and persist this to db along with the s3 url and credentials
- Created placeholder endpoint for optics to call when everything is implemented

### CODE REVIEW

File: app/jobs/send_complaint_job.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@ class SendComplaintJob < ApplicationJob
 
   def perform(form_builder_payload:)
     Rails.logger.warn("Working on job_id: #{job_id}")
+
+    Usecase::SpawnAttachments.new(form_builder_payload: form_builder_payload).call
```
Comment (tommotaylor) [inline app/jobs/send_complaint_job.rb]: I thought we would "spawn" the attachments when we receive the initial `POST` request from the submitter. Eg:
1. Controller receives POST
2. Controller iterates through the attachments array and saves each one to the db
3. Controller invokes the `SendComplaintJob`, sending the complaint to Optics

Comment (asmega) [reply app/jobs/send_complaint_job.rb]: i considered that, but why do it then when we can do it async? I don't see why the API client has to wait

File: app/lib/usecase/download_attachment.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require 'net/http'
+
+module Usecase
+  class DownloadAttachment
+    def initialize(identifier:)
+      @identifier = identifier
+    end
+
+    def call
+      Net::HTTP.get(uri)
```
[the remark this answers is not in the document]
Comment (asmega) [reply app/lib/usecase/download_attachment.rb]: I can do the error handling

What's the benefit of using HTTPart over ruby stdlib?

File: app/lib/usecase/spawn_attachment.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+module Usecase
+  class SpawnAttachment
+    def initialize(klass: Attachment, params:)
```
[the remark this answers is not in the document]
Comment (asmega) [reply app/lib/usecase/spawn_attachment.rb]: i've now removed class injection from the class

**Pull request merged**